ArduCopter/control_guided.cpp

```diff
@@ -663,6 +663,14 @@ void Copter::guided_set_desired_velocity_with_accel_and_fence_limits(const Vecto
 
     // exit immediately if already equal
     if (curr_vel_des == vel_des) {
```

Comment:
My proposal for this was:

```
if (curr_vel_des != vel_des) {

    // get change in desired velocity
    Vector3f vel_delta = vel_des - curr_vel_des;

    [...]
}

#if AC_AVOID_ENABLED
[...]
```

I see no reason to run the delta math calculations when delta will be zero.

Reply:
Yes @OXINARF , it definitely makes sense. @rmackay9 eagerly awaiting your response.